powa_collector/powa_worker.py

```diff
@@ -400,10 +400,10 @@ def __worker_main(self):
                     self.logger.error("Error during snapshot: %s" % e)
                     if (self.__repo_conn is None
                             or self.__repo_conn.closed > 0):
-                        self.__repo_conn = None
+                        self.__disconnect_repo()
                     if (self.__remote_conn is None
                             or self.__remote_conn.closed > 0):
-                        self.__remote_conn = None
+                        self.__disconnect_repo()
```

Comment:
This should handle the remote connection :)

Reply:
It's not closing the correct one?

Reply:
Yes, __disconnect_repo() only closes the repository connection.  There's no function to close the remote connection as it's supposed to be a persistent connection.

Reply:
Fixed this silly error :)